okhttp/src/main/java/com/squareup/okhttp/internal/DiskLruCache.java

```diff
@@ -772,14 +772,15 @@ private void validateKey(String key) {
     private Snapshot(String key, long sequenceNumber, Source[] sources, long[] lengths) {
       this.key = key;
       this.sequenceNumber = sequenceNumber;
-      this.sources = sources;
-      this.lengths = lengths;
+      this.sources = sources == null ? null : sources.clone();
+      this.lengths = lengths == null ? null : lengths.clone();
```

Comment:
Similarly, `Sources[]` is created specifically for this constructor call and `lengths` is defensively cloned in the calling code.